src/components/SignUp/Group/index.tsx

```diff
@@ -0,0 +1,244 @@
+import { createUserWithEmailAndPassword } from 'firebase/auth';
+import { addDoc, collection } from 'firebase/firestore';
+import React, { useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { Step, useForm, useStep } from 'react-hooks-helper';
+import { useFirebaseContext } from '../../../context/FirebaseContext';
+import { useProfileContext } from '../../../context/ProfileContext';
+import Profile from '../../../pages/Profile';
+import PageContainer from '../../layout/PageContainer';
+import ActorInfo1 from '../Individual/ActorInfo1';
+import ActorInfo2 from '../Individual/ActorInfo2';
+import IndividualBasics from '../Individual/Basics';
+import Demographics from '../Individual/Demographics';
+import OffstageRoles from '../Individual/OffstageRoles';
+import ProfilePhoto from '../Individual/ProfilePhoto';
+import Privacy from '../Privacy';
+import SignUpFooter from '../SignUpFooter';
+
+const GroupSignUp: React.FC<{
+  currentStep: number;
+  setCurrentStep: (x: number) => void;
+}> = ({ currentStep, setCurrentStep }) => {
+  const { firebaseAuth, firebaseFirestore } = useFirebaseContext();
+  const { setAccountRef, setProfileRef } = useProfileContext();
+  const [formData, setForm] = useForm(defaultData); // useForm is an extension of React hooks to manage form state
+
+  // default state for form validation error states per step
+  const [stepErrors, setStepErrors] = useState(
+    createDefaultStepErrorsObj(flatSteps(defaultSteps))
+  );
+
+  // console.log('formData', { formData });
+  // defaults for our defaultSteps
+  const { step, index, navigation } = useStep({
+    initialStep: currentStep,
+    steps: defaultSteps
+  });
+  const stepId = (step as Step).id;
```

Comment:
I feel like we can extract some of this logic we have in all of our SignUp steps like individual 1, individual 2, and now group and some generic utilities for it 🤔

Reply:
Yep I will extract more.